src/psyclone/tests/parse/kernel_test.py

```diff
@@ -0,0 +1,272 @@
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors R. W. Ford and A. R. Porter, STFC Daresbury Lab
+
+'''A module to perform pytest unit tests on the parse/kernel.py
+file. Some tests for this file are in parse_test.py. This file adds
+tests for code that is not covered there.'''
+
+import os
+import pytest
+from psyclone.parse.kernel import KernelType, get_kernel_metadata, \
+    KernelProcedure, Descriptor, BuiltInKernelTypeFactory, get_kernel_filepath
+from psyclone.parse.utils import ParseError
+from fparser.api import parse
+
+# pylint: disable=invalid-name
+
+
+CODE = (
+    "module test_mod\n"
+    "  type, extends(kernel_type) :: test_type\n"
+    "    type(arg_type), dimension(1) :: meta_args =    &\n"
+    "          (/ arg_type(gh_field,gh_write,w1) /)\n"
+    "     integer :: iterates_over = cells\n"
+    "   contains\n"
+    "     procedure, nopass :: code => test_code\n"
+    "  end type test_type\n"
+    "contains\n"
+    "  subroutine test_code()\n"
+    "  end subroutine test_code\n"
+    "end module test_mod\n"
+
+    )
+
+# function get_kernel_filepath
+
+
+def test_getkernelfilepath_nodir():
+    '''Test that an appropriate exception is raised if the specified
+    directory does not exist.
+
+    '''
+    with pytest.raises(ParseError) as excinfo:
+        _ = get_kernel_filepath("test_mod", "non/existant/file/path", None)
+    assert ("Supplied kernel search path does not exist or cannot be "
+            "read") in str(excinfo.value)
+
+
+def test_getkernelfilepath_multifile(tmpdir):
+    '''Test that an appropriate exception is raised if more than one file
+    matches when searching for kernels.
+
+    '''
+    filename = str(tmpdir.join("test_mod.f90"))
+    ffile = open(filename, "w")
+    ffile.write("")
+    ffile.close()
+    os.mkdir(str(tmpdir.join("tmp")))
+    filename = str(tmpdir.join("tmp", "test_mod.f90"))
+    ffile = open(filename, "w")
+    ffile.write("")
+    ffile.close()
+
+    with pytest.raises(ParseError) as excinfo:
+        _ = get_kernel_filepath("test_mod", str(tmpdir), None)
+    assert ("More than one match for kernel file 'test_mod.[fF]90' "
+            "found!") in str(excinfo.value)
+
+# class BuiltInKernelTypeFactory():create test
+
+
+def test_builtinfactory_metadataerror(monkeypatch):
+    '''Test that an appropriate exception is raised if the builtin
+    metadata cant be parsed. This is difficult to trigger so use
+    monkeypatch.
+
+    '''
+    from psyclone.dynamo0p3_builtins import BUILTIN_MAP as builtins
+    from psyclone.dynamo0p3_builtins import BUILTIN_DEFINITIONS_FILE as \
+        fname
+    from fparser import api as fpapi
+    # Use 1/0 to raise an exception in the lambda function.
```

Comment:
I'm not keen on this use of 1/0 to generate an exception :-) Please could you define a separate function that explicitly raises an Exception and then use that to monkeypatch the `parse` method.

Reply:
I only need to cause an exception so I replaced the function with None which causes a typeerror as None is not callable. I've mentioned this in the docstring.

Reply:
OK, thanks.